rueidisotel/metrics.go

```diff
@@ -0,0 +1,163 @@
+package rueidisotel
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"sync/atomic"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/metric"
+
+	"github.com/redis/rueidis"
+)
+
+var (
+	DefaultDialLatencyHistogramDefaultBuckets = []float64{
+		.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10,
+	}
+	DefaultDialFn = func(dst string, dialer *net.Dialer, _ *tls.Config) (conn net.Conn, err error) {
+		return dialer.Dial("tcp", dst)
+	}
+)
+
+type DialLatencyHistogramOption struct {
```

Comment:
```suggestion
type HistogramOption struct {
```
I believe we can reuse this option struct for other histogram metrics. So, name it just `HistogramOption` may be better.

Reply:
[3895581](https://github.com/redis/rueidis/pull/421/commits/3895581d888aebcbc00fbff76a97e71add2c317d)